Documentation/architecture.md

```diff
@@ -0,0 +1,68 @@
+# Rocket architecture
+
+## Overview
+
+Rocket consists only of a command-line tool, `rkt`, and does not have a daemon. All state is communicated via the filesystem. Facilities like file-locking are used to ensure co-operation and mutual exclusion between concurrent invocations of the `rkt` command.
```

Comment:
Can you add a few words about _why_ the lack of a daemon is important. Probably needs to touch on security, compostability, etc.